superset-frontend/src/components/Chart/DrillDetail/DrillDetailMenuItems.tsx

```diff
@@ -0,0 +1,230 @@
+        &.anticon {
+          font-size: unset;
+          .anticon {
+            line-height: unset;
+            vertical-align: unset;
+          }
+        }
+      `}
+    />
+  </Tooltip>
+);
+
+const DisabledMenuItem = ({ children, ...props }: { children: ReactNode }) => (
+  <Menu.Item disabled {...props}>
+    <div
+      css={css`
+        white-space: normal;
+        max-width: 160px;
+      `}
+    >
+      {children}
+    </div>
+  </Menu.Item>
+);
+
+const Filter = styled.span`
+  ${({ theme }) => `
+     font-weight: ${theme.typography.weights.bold};
+     color: ${theme.colors.primary.base};
+   `}
+`;
+
+export type DrillDetailMenuItemsProps = {
+  chartId: number;
+  formData: QueryFormData;
+  filters?: BinaryQueryObjectFilterClause[];
+  isContextMenu?: boolean;
+  onSelection?: () => void;
+  onClick?: (event: MouseEvent) => void;
+};
+
+const DrillDetailMenuItems = ({
+  chartId,
+  formData,
+  filters = [],
+  isContextMenu = false,
+  onSelection = () => null,
+  onClick = () => null,
+  ...props
+}: DrillDetailMenuItemsProps) => {
+  const [modalFilters, setFilters] = useState<BinaryQueryObjectFilterClause[]>(
+    [],
+  );
+
+  const [showModal, setShowModal] = useState(false);
+  const openModal = useCallback(
+    (filters, event) => {
+      onClick(event);
+      onSelection();
+      setFilters(filters);
+      setShowModal(true);
+    },
+    [onClick, onSelection],
+  );
+
+  const closeModal = useCallback(() => {
+    setShowModal(false);
+  }, []);
+
+  // Check for Behavior.DRILL_TO_DETAIL to tell if plugin handles the `contextmenu`
+  // event for dimensions.  If it doesn't, tell the user that drill to detail by
+  // dimension is not supported.  If it does, and the `contextmenu` handler didn't
+  // pass any filters, tell the user that they didn't select a dimension.
+  const handlesDimensionContextMenu = useMemo(
+    () =>
+      getChartMetadataRegistry()
+        .get(formData.viz_type)
+        ?.behaviors.find(behavior => behavior === Behavior.DRILL_TO_DETAIL),
+    [formData.viz_type],
+  );
+
+  // Check chart plugin metadata to see if chart's current configuration lacks
+  // aggregations, in which case Drill to Detail should be disabled.
+  const noAggregations = useMemo(
+    () =>
+      getChartMetadataRegistry()
+        .get(formData.viz_type)
+        ?.noAggregations?.(formData) ?? false,
+    [formData],
+  );
+
+  const drillToDetail = noAggregations ? (
+    <DisabledMenuItem {...props} key="drill-detail-no-aggregations">
+      {t('Drill to detail')}
+      <DisabledMenuItemTooltip
+        title={t(
+          'Drill to detail is disabled because this chart does not group data by dimension value.',
+        )}
+      />
+    </DisabledMenuItem>
+  ) : (
+    <Menu.Item
+      {...props}
+      key="drill-detail-no-filters"
+      onClick={openModal.bind(null, [])}
+    >
+      {t('Drill to detail')}
+    </Menu.Item>
+  );
+
+  let drillToDetailBy;
```

Comment:
this one is the `menuItem` component so should rename to `menuItem` or `drillDetailMenuItem`.

Reply:
Same as above.

Reply:
Ok, renamed `drillToDetail` and `drillToDetailBy` to `drillToDetailMenuItem` and `drillToDetailByMenuItem`.